Add RGB, RGBW and RGBWW capability to template.light

Vaarlion opened the pull request:
## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->
This clean old mention of the white control that no-longer work since https://github.com/home-assistant/core/pull/76923 and add the new documentation for it's replacement, RGB, RGBW and RGBWW control.

It also replace the still working but deprecated `color` keyword for the new and more accurate `hs` keyword.


## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [X] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [X] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/86047
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [X] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [X] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

Vaarlion committed:
Remove deprecate 'white' settings and replace them with new RGBWW settings

YuriiMaiboroda commented on source/_integrations/light.template.markdown:
```diff
@@ -165,12 +170,21 @@ light:
+      set_rgbww:
+        description: "Defines an action to run when the light is given a rgbww color command. Available variables: `rgbww` as a tuple, `rgb` as a tuple, `r`, `g`, `b`, `cw` and `ww`"
+        required: false
         required: false
```
`required: false` is duplicated

Vaarlion replied:
Good catch :)

YuriiMaiboroda reviewed (changes requested):
Need to remove duplicate of `required: false` for `set_rgbww`.

Vaarlion committed:
Cleanup duplicated line

https://github.com/home-assistant/home-assistant.io/pull/25994#discussion_r1341976638

YuriiMaiboroda reviewed (approved):
LGTM

Vaarlion commented:
rebased one last time, i no longer working on this.
It's been to long.

c0ffeeca7 commented on source/_integrations/light.template.markdown:
```diff
@@ -105,13 +100,23 @@ light:
-      white_value_template:
-        description: Defines a template to get the white value of the light.
+      hs_template:
+        description: Defines a template to get the hs color of the light. Must render a tuple (hue, saturation)
```
```suggestion
        description: Defines a template to get the HS color of the light. Must render a tuple (hue, saturation).
```

c0ffeeca7 commented on source/_integrations/light.template.markdown:
```diff
@@ -105,13 +100,23 @@ light:
+        type: template
+        default: optimistic
+      rgb_template:
+        description: Defines a template to get the rgb color of the light. Must render a tuple or a list (red, green, blue)
```
```suggestion
        description: Defines a template to get the RGB color of the light. Must render a tuple or a list (red, green, blue).
```

c0ffeeca7 commented on source/_integrations/light.template.markdown:
```diff
@@ -105,13 +100,23 @@ light:
+        type: template
+        default: optimistic
+      rgbw_template:
+        description: Defines a template to get the rgbw color of the light. Must render a tuple or a list (red, green, blue, white)
```
```suggestion
        description: Defines a template to get the RGBW color of the light. Must render a tuple or a list (red, green, blue, white).
```

c0ffeeca7 commented on source/_integrations/light.template.markdown:
```diff
@@ -105,13 +100,23 @@ light:
-      color_template:
-        description: Defines a template to get the color of the light. Must render a tuple (hue, saturation)
+      rgbww_template:
+        description: Defines a template to get the rgbww color of the light. Must render a tuple or a list (red, green, blue, cold white, warm white)
```
```suggestion
        description: Defines a template to get the RGBWW color of the light. Must render a tuple or a list (red, green, blue, cold white, warm white).
```

c0ffeeca7 commented on source/_integrations/light.template.markdown:
```diff
@@ -165,12 +170,20 @@ light:
-      set_color:
-        description: Defines an action to run when the light is given a color command.
+      set_rgb:
+        description: "Defines an action to run when the light is given a rgb color command. Available variables: `rgb` as a tuple, `r`, `g` and `b`"
```
```suggestion
        description: "Defines an action to run when the light is given an RGB color command. Available variables: `rgb` as a tuple, `r`, `g` and `b`."
```

c0ffeeca7 commented on source/_integrations/light.template.markdown:
```diff
@@ -165,12 +170,20 @@ light:
+        required: false
+        type: action
+      set_rgbw:
+        description: "Defines an action to run when the light is given a rgbw color command. Available variables: `rgbw` as a tuple, `rgb` as a tuple, `r`, `g`, `b` and `w`"
```
```suggestion
        description: "Defines an action to run when the light is given an RGBW color command. Available variables: `rgbw` as a tuple, `rgb` as a tuple, `r`, `g`, `b` and `w`."
```

c0ffeeca7 commented on source/_integrations/light.template.markdown:
```diff
@@ -165,12 +170,20 @@ light:
+        required: false
+        type: action
+      set_rgbww:
+        description: "Defines an action to run when the light is given a rgbww color command. Available variables: `rgbww` as a tuple, `rgb` as a tuple, `r`, `g`, `b`, `cw` and `ww`"
```
```suggestion
        description: "Defines an action to run when the light is given an RGBWW color command. Available variables: `rgbww` as a tuple, `rgb` as a tuple, `r`, `g`, `b`, `cw` and `ww`."
```

c0ffeeca7 commented on source/_integrations/light.template.markdown:
```diff
@@ -194,8 +207,8 @@ For example, you would replace
-Transition doesn't have its own script, it will instead be passed as a named parameter `transition` to the `turn_on`, `turn_off`, `brightness`, `color_temp`, `effect`, `hs_color` or `white_value` scripts.
-Brightness will be passed as a named parameter `brightness` to either of `turn_on`, `color_temp`, `effect`, `hs_color` or `white_value` scripts if the corresponding parameter is also in the call. In this case the brightness script (`set_level`) will not be called.
+Transition doesn't have its own script, it will instead be passed as a named parameter `transition` to the `turn_on`, `turn_off`, `brightness`, `color_temp`, `effect`, `hs_color`, `rgb_color`, `rgbw_color` or `rgbww_color` scripts.
+Brightness will be passed as a named parameter `brightness` to either of `turn_on`, `color_temp`, `effect`, `hs_color`, `rgb_color`, `rgbw_color` or `rgbww_color` scripts if the corresponding parameter is also in the call. In this case the brightness script (`set_level`) will not be called.
```
```suggestion
Brightness will be passed as a named parameter `brightness` to either of `turn_on`, `color_temp`, `effect`, `hs_color`, `rgb_color`, `rgbw_color` or `rgbww_color` scripts if the corresponding parameter is also in the call. In this case, the brightness script (`set_level`) will not be called.
```

c0ffeeca7 commented on source/_integrations/light.template.markdown:
```diff
@@ -351,3 +364,58 @@ light:
 
 {% endraw %}
+
+### Make a Global light entity for a multi segments WLED light
```
```suggestion
### Make a global light entity for a multi-segment WLED light
```

c0ffeeca7 committed:
tiny tweaks

c0ffeeca7 reviewed (dismissed):
Thank you, @Vaarlion :+1:

c0ffeeca7 committed:
Merge branch 'next' into feature/80449_light_template_RGBWW
